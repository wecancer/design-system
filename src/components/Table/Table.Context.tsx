import React from 'react'

type TableContext = {
  cellsWidth: string[]
  setCellsWidth: React.Dispatch<React.SetStateAction<string[]>>
}

const defaultValue: TableContext = {
  cellsWidth: [],
  setCellsWidth: () => null,
}

export default React.createContext<TableContext>(defaultValue)
