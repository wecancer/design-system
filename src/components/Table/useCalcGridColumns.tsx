import {useContext} from 'react'
import TableContext from './Table.Context'

export default function useCalcGridColumns(collsLength: number): string {
  const {cellsWidth} = useContext(TableContext)

  return new Array(collsLength)
    .fill(null)
    .map((item, index) => cellsWidth[index] || '1fr')
    .join(' ')
}
