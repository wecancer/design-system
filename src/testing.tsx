import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
