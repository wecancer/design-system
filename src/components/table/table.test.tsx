import Table from '.'
import TableRow from './row'
import TableCell from './cell'
import TableHeader from './header'
import { render } from '../../testing'

describe('<Table />', () => {
  it('should render snapshot correctly', () => {
    const { container } = render(
      <Table cellsWitdh="100px 200px 1fr 1fr">
        <TableHeader>
          <TableCell>Head 1</TableCell>
          <TableCell>Head 2</TableCell>
          <TableCell>Head 3</TableCell>
          <TableCell>Head 4</TableCell>
        </TableHeader>
        <TableRow onClick={() => null}>
          <TableCell type="warning">Warning</TableCell>
          <TableCell>Clickable</TableCell>
          <TableCell>Item 3</TableCell>
          <TableCell>
            Proin viverra justo vitae aliquam pulvinar. Interdum et malesuada
            fames ac ante ipsum primis in faucibus.
          </TableCell>
        </TableRow>
        <TableRow>
          <TableCell type="error">Error!</TableCell>
          <TableCell>Item 2</TableCell>
          <TableCell>Item 3</TableCell>
          <TableCell>
            Phasellus vel velit et mi eleifend dignissim. Duis vitae mi magna.
            Proin in feugiat urna.
          </TableCell>
        </TableRow>
        <TableRow>
          <TableCell type="input">Item 1</TableCell>
          <TableCell>Item 2</TableCell>
          <TableCell>Item 3</TableCell>
          <TableCell>
            Orci varius natoque penatibus et magnis dis parturient montes,
            nascetur ridiculus mus.
          </TableCell>
        </TableRow>
        <TableRow>
          <TableCell>Item 1</TableCell>
          <TableCell>Item 2</TableCell>
          <TableCell>Item 3</TableCell>
          <TableCell>
            Nulla iaculis, erat ultrices egestas pretium, metus augue efficitur
            massa, blandit elementum arcu dolor at erat.
          </TableCell>
        </TableRow>
        <TableRow>
          <TableCell>Item 1</TableCell>
          <TableCell>Item 2</TableCell>
          <TableCell>Item 3</TableCell>
          <TableCell>Item 4</TableCell>
        </TableRow>
        <TableRow>
          <TableCell>Item 1</TableCell>
          <TableCell>Item 2</TableCell>
        </TableRow>
      </Table>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
