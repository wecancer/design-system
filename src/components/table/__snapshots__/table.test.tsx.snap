// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render snapshot correctly 1`] = `
<ul
  class="sc-bdvvtL cvfeBv"
>
  <li
    class="sc-eCImPb dfihNe"
  >
    <div
      class="sc-jRQBWg efedTK"
    >
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Head 1
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Head 2
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Head 3
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Head 4
        </div>
      </div>
    </div>
  </li>
  <li
    class="sc-gsDKAQ cVUWzY"
    role="button"
  >
    <div
      class="sc-dkPtRN oQTeg"
    >
      <div
        class="sc-hKwDye hKxfRJ wc-table-cell"
      >
        <div>
          Warning
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Clickable
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 3
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Proin viverra justo vitae aliquam pulvinar. Interdum et malesuada fames ac ante ipsum primis in faucibus.
        </div>
      </div>
    </div>
  </li>
  <li
    class="sc-gsDKAQ UKGbb"
  >
    <div
      class="sc-dkPtRN oQTeg"
    >
      <div
        class="sc-hKwDye hKxfRJ wc-table-cell"
      >
        <div>
          Error!
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 2
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 3
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Phasellus vel velit et mi eleifend dignissim. Duis vitae mi magna. Proin in feugiat urna.
        </div>
      </div>
    </div>
  </li>
  <li
    class="sc-gsDKAQ bcAwMB"
  >
    <div
      class="sc-dkPtRN oQTeg"
    >
      <div
        class="sc-hKwDye hKxfRJ wc-table-cell"
      >
        <div>
          Item 1
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 2
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 3
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.
        </div>
      </div>
    </div>
  </li>
  <li
    class="sc-gsDKAQ fNypaA"
  >
    <div
      class="sc-dkPtRN oQTeg"
    >
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 1
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 2
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 3
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Nulla iaculis, erat ultrices egestas pretium, metus augue efficitur massa, blandit elementum arcu dolor at erat.
        </div>
      </div>
    </div>
  </li>
  <li
    class="sc-gsDKAQ fNypaA"
  >
    <div
      class="sc-dkPtRN oQTeg"
    >
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 1
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 2
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 3
        </div>
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        <div>
          Item 4
        </div>
      </div>
    </div>
  </li>
  <li
    class="sc-gsDKAQ iRVJcu"
  >
    <div
      class="sc-dkPtRN dGrlTs"
    >
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        Item 1
      </div>
      <div
        class="sc-hKwDye eDPjSw wc-table-cell wc-table-cell-solid"
      >
        Item 2
      </div>
    </div>
  </li>
</ul>
`;
